prow/tide/tide.go

```diff
@@ -1163,12 +1174,17 @@ func (c *Controller) presubmitsByPull(sp *subpool) (map[int][]config.Presubmit,
 		}
 	}
 
-	for _, ps := range c.config().Presubmits[sp.org+"/"+sp.repo] {
-		if !ps.ContextRequired() {
-			continue
+	for _, pr := range sp.prs {
+		presubmits, err := c.config().GetPresubmits(c.gc, sp.org+"/"+sp.repo, refGetterFactory(sp.sha), refGetterFactory(string(pr.HeadRefOID)))
```

Comment:
Why does `GetPresubmits` use getter functions for the shas? When do we call `GetPresubmits` without already having the SHAs available?

Reply:
Never in Tide but in all other current users for `GetPresubmits`:
* The `skip` plugin
* The `trigger` plugin
* In ` Deck